### DISCUSSION TITLE
Some Problem of ESP 32C3

### DISCUSSION THREAD

**Discussion by PastGone** [micropython/micropython#11178] [ESP32]
i want to use I2s on my esp32c3 so  i try to 

```python 
from machine import I2S
```
 
but some erro happened
the shell says
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: can't import name I2S 
```
what should i do?

**Comment by puppet13th** [comment]
currently I2S not enabled for esp32-c3 firmware. https://github.com/micropython/micropython/blob/master/ports/esp32/boards/GENERIC_C3/mpconfigboard.h

**Reply by PastGone** [reply]
What could I do,and how can i build a firmware with I2S?Did you have some video or class to share?Thank you very much.

**Comment by ricksorensen** [comment]
I have gotten I2S to work on an ESP32C3 (SEEED XIAO) for a small sample.  Thanks to Mike Alexander [https://github.com/micropython/micropython/issues/11245](https://github.com/micropython/micropython/issues/11245) for finding a patch for the static sound.
[https://github.com/micropython/micropython/issues/8641](https://github.com/micropython/micropython/issues/8641)

Here is my repository with the i2s mods for esp32c3: [https://github.com/ricksorensen/micropython/tree/esp32c_i2s](https://github.com/ricksorensen/micropython/tree/esp32c_i2s) in the `esp32c_i2s` branch.  Note that this assumes building with ESP IDF v5 SDK.

I have since tried this out on UM_TINYS3 and GENERIC_S3_SPIRAM_OCT with Mike Teachman's `easy_wav_player` with both stereo and mono wav files and this patch worked fine.  I still want to figure out how to convert from the v4 SDK the to current ESP IDF v5 I2S api.

**Reply by PastGone** [reply]
thank you,i will try to use this on my esp32,and i hope everything will be ok.

**Comment by bidrohini** [comment]
Here is an archived thread discussing the same error message: https://forum.micropython.org/viewtopic.php?t=12384&p=67173
